### DISCUSSION TITLE
Only first part of renderable displayed?

### DISCUSSION THREAD

**Discussion by The-Compiler** [Textualize/textual#123] [Q&A]
Hey! I'm trying to use Textual for the first time, with the aim of integrating [jut](https://github.com/kracekumar/jut) (which already uses Rich) to display both Jupyter files and associated test results.

Unfortunately, jut doesn't directly provide a Rich renderable, but I wrote a small adapter class which does so. Using Rich directly prints the output as expected:

![image](https://user-images.githubusercontent.com/625793/135264556-8644ff19-a835-47a4-8bf8-bf07b81de9c5.png)

However, when trying to integrate it with Textual in a scrolling view, only the panes from the first Jupyter cell are displayed - after that I just get an empty scroll view:

![image](https://user-images.githubusercontent.com/625793/135264392-51e7d4a8-5d2c-4cef-9d53-683dc634833f.png)

Here's my code:

```python
import jut
import rich.console
import textual.app
import textual.widgets
import textual.events


NOTEBOOK = "../development/notebooks/1-getting-started/01-python-basic/python-basics.ipynb"


class JutRenderable:

    def __init__(self, filename: str) -> None:
        jut_config = jut.Config(
            input_file=filename,
            head=10,  # ignored due to full_display=True
            tail=None,
            single_page=True,
            full_display=True,
            force_colors=False,
            start=None,
            end=None,
            exclude_output_cells=False,
            no_cell_border=False,
        )
        self.render = jut.Render(jut_config)
        self.render.parse_notebook()
    
    # FIXME __rich__ vs. __rich_console__?
    def __rich_console__(self, *args, **kwargs) -> rich.console.RenderResult:
        for index, cell in self.render.iter_cells():
            yield from self.render.format_cell(index, cell)


class App(textual.app.App):

    async def on_mount(self, event: textual.events.Mount) -> None:
        self.source = textual.widgets.ScrollView()
        self.result = textual.widgets.Placeholder()
        await self.view.dock(textual.widgets.Footer(), edge="bottom")
        grid = await self.view.dock_grid()
        grid.add_column("left")
        grid.add_column("right")
        grid.add_row("row")

        grid.place(self.source, self.result)

        await self.call_later(self.render_notebook, NOTEBOOK)

    async def render_notebook(self, filename: str) -> None:
        renderable = JutRenderable(filename)
        await self.source.update(renderable)
    

if True:
    App.run(title="AutPy Commander")
else:
    renderable = JutRenderable(NOTEBOOK)
    rich.print(renderable)
```

(Note: If you want to reproduce it, you'll need to fight with pip a bit, as unfortunately `jut` declares an older Rich version in it's `pyproject.toml` which is incompatible with Textual - but the newest Rich version works just fine.)

Am I doing something obvious wrong there, or is this a bug? If it's the latter, I might try to get a more minimal reproducer.

**Comment by paw-lu** [comment]
hi @The-Compiler! did anything come of this?

I'm looking to do something similar, but with [paw-lu/nbpreview](https://github.com/paw-lu/nbpreview) (which is _extremely_ similar to Jut).
